src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -258,3 +264,117 @@ load_events_from_file(FILE* fptr)
     return parsed_events;
 
 }
+
+
+int dump_file_info(PyObject* file_info, FILE* fptr){
+    const char * buffer = NULL;
+    uint64_t content_length = 0;
+    uint64_t compression_length = 0;
+    PyObject* json_ret = NULL;
+    PyObject* zlib_ret = NULL;
+    PyObject* bytes_data = NULL;
+    PyObject* dumps_func = PyObject_GetAttrString(json_module, "dumps");
+    PyObject* json_args = PyTuple_New(1);
+    PyObject* compress_func = PyObject_GetAttrString(zlib_module, "compress");
+    PyObject* zlib_args = PyTuple_New(1);
+
+    if (!dumps_func) {
+        perror("Failed to access json.dumps()");
```

Comment:
Do not do `perror` then `exit`. This will confuse Python users as their program will just exit. It's hard to debug and it's impossible to catch such errors in Python. Check the error handling from other code. Do `PyErr_SetString` and `goto` a `cleanup` label(where resources are released in case of an error).